woocommerce/woocommerce-android
# [Coupons] Coupon Summary screen UI

Part of #5529, #5531


### Description


This implements the coupon summary screen design below, with some details* and exception**:

<img src="https://user-images.githubusercontent.com/266376/160364218-0ff56766-8ef0-4efd-8b62-85a53457eeff.png" width="35%">

[*] **Details**
Added items:
- Fragment, Compose components, and coupon summary view model
- Navigation from Coupon List screen to Coupon Summary screen
- All items displayed on the Coupon Summary screen are demo data

[*] **Exception**
- I added some simple parsing from demo data in viewmodel on the various Compose components. However, some items like expiration date data and the whole Performance section are hardcoded. To me they felt like they will require some non-trivial amount of work to parse and display properly, so I opted to focus on just the design for now.
- Triple-dot menu on top right. The menu is meant for various actions, refer to #5533 for starters.
- The app bar does not show the coupon name yet (says "Coupons" instead).  

### Testing instructions


1. Run app, go to Menu tab, then tap the Coupons button. This should show the Coupons list screen,
2. Tap any shown coupon and it should show the Coupon Summary screen. Check and compare the shown UI with the Figma design.

### Images/gif


| Dark | Light |
|-|-|
| ![Dark version of Coupon Summary UI](https://user-images.githubusercontent.com/266376/160369279-cc3cf1a0-acc7-4666-b555-77306bc71ff2.png) | ![Light version of Coupon Summary UI](https://user-images.githubusercontent.com/266376/160369276-1d65dea7-d020-48c0-920a-09458f493903.png) |
